+

makivka opened the pull request.

makivka committed:
+

makivka committed:
output in calculation commented

makivka committed:
main program is edited
Draw methods are edited

ParadoxMH commented on Program.cs:
```diff
@@ -6,7 +6,32 @@ class Program
+            Shape[] shapes = new Shape[3];
+            shapes[0] = circle1;
+            shapes[1] = rectangle1;
+            shapes[2] = circle2;
```
As `circle1`, `rectangle1`, and `circle2` are used only 1 time, you can init them right here

makivka replied:
done

ParadoxMH commented on Program.cs:
```diff
@@ -6,7 +6,32 @@ class Program
+            shapes[0] = circle1;
+            shapes[1] = rectangle1;
+            shapes[2] = circle2;
+            for (int i = 0; i < shapes.Length; i++)
```
how about foreach?)

makivka replied:
done

ParadoxMH commented on Program.cs:
```diff
@@ -6,7 +6,32 @@ class Program
+                Radius = 6,
+                Coordinates = new double[6, 2] { { 9, 9 }, { 8, 8 }, { 7, 7 }, { 6, 6 }, { 5, 5 }, { 4, 4 } }
+            };
+            Shape[] shapes = new Shape[3];
```
Can you use List<Shape> here?
What is the difference between using a list and an array?

makivka replied:
An array has a fixed dimension, and a list we use, when the number of elements is unknown at the beginning

makivka replied:
done

ParadoxMH commented on Circles.cs:
```diff
@@ -0,0 +1,40 @@
+
+namespace Figures
+{
+    class Circles : Shape
```
I suggest creating a constructor here so we could avoid using the `Radius` property.
You'll pass radius into the constructor and you'll call `CircleCoordinates()` method directly from the constructor.

ParadoxMH replied:
and lets add `double radius` as parameter for the `CircleCoordinates `

makivka replied:
done

ParadoxMH commented on Circles.cs:
```diff
@@ -0,0 +1,40 @@
+            }
+
+
+        }
```
clean up useless whitespaces please

makivka replied:
done

makivka committed:
List instead of array
circle1, rectangle1, and circle2 are inited  right in List
"foreach" instead of "for" with List
Constructors are created for Shape, Circle, Rectangle
 useless whitespaces are cleaned up

makivka committed:
void Draw() for Circless class updated

makivka committed:
void Draw() for Rectangle updated
